### DISCUSSION TITLE
EmphasizeElements does not seem to work on "some" iModels

### DISCUSSION THREAD

**Discussion by candcconsulting** [iTwin/itwinjs-core#2288] [Q&A] [answered]
Can anyone advise where to start looking for the below issue I have
When I run a rule checker on a specific iModel and the failed results are emphasized all works as expected
![image](https://user-images.githubusercontent.com/19268085/133965343-82368d19-a3b5-4dda-9d9f-951ff8de1f53.png)
```
    let emph = EmphasizeElements.getOrCreate(IModelApp.viewManager.selectedView!);      
    emph.wantEmphasis = true;  
    //IModelApp.viewManager.selectedView!.zoomToElements(invalidElements, { animateFrustumChange: true, standardViewId: StandardViewId.RightIso});
    emph.emphasizeElements(invalidElements, IModelApp.viewManager.selectedView!, undefined , true)

```
However, when I run a different rule on different iModel then the emphasize does not work and there are no errors in console
Even though the second rule only adds about 780 elements to the emphasize list
![image](https://user-images.githubusercontent.com/19268085/133965484-e6089e0d-6543-40ac-b871-cc9f2001981a.png)

If I try and step into the emphasizeElements, Chrome reports Debugger paused
I can add people to the project if required, but I am hoping the same issue happens on other iModels

**Comment by pmconne** [comment, marked as answer]
In the case where emphasizeElements is having no effect, does the zoomToElements call still correctly zoom to the elements?
Are you opening the iModels one after the other in a single instance of the app? If so, does the second iModel work as expected if you restart the app and open that iModel first?

**Reply by candcconsulting** [reply]
Thanks Paul,
I uncommented the zoomToElements and the model that highlights works (ie zooms), but the model that does not work does not zoom.  Let me build some more error checking into the original rule so I can run it on other models as I have some undefined results which need to be managed ...

**Reply by candcconsulting** [reply]
Thanks Paul, there was an error in my second rule SQL which was finding elements, but not geometric elements.
![image](https://user-images.githubusercontent.com/19268085/134138897-1bc8ea6b-4031-4458-b3c3-0ff7d0316a3b.png)
